src/Sulu/Component/Content/Metadata/Loader/XmlLegacyLoader.php

```diff
@@ -137,12 +137,21 @@ private function loadTemplateAttributes($resource, \DOMXPath $xpath, $type)
                 'view' => $this->getValueFromXPath('/x:template/x:view', $xpath),
                 'controller' => $this->getValueFromXPath('/x:template/x:controller', $xpath),
                 'internal' => $this->getValueFromXPath('/x:template/x:internal', $xpath),
-                'cacheLifetime' => $this->getValueFromXPath('/x:template/x:cacheLifetime', $xpath),
+                'cacheLifetime' => intval($this->getValueFromXPath('/x:template/x:cacheLifetime', $xpath)),
                 'tags' => $this->loadStructureTags('/x:template/x:tag', $xpath),
                 'meta' => $this->loadMeta('/x:template/x:meta/x:*', $xpath),
             );
 
-            $result = array_filter($result);
+            $result = array_filter(
+                $result,
+                function ($value) {
+                    if (is_array($value)) {
+                        return $value != null;
+                    }
```

Comment:
This check is really confusing... I guess you also want to check if the array is empty. Isn't it possible to just use `$value != null` for the entire callback? Or to use `empty()`? And to write that in a single line?